File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/CS/Direct2D/D2DShapes/LineShape.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using Microsoft.WindowsAPICodePack.DirectX.Direct2D1;

namespace D2DShapes
{
    internal class LineShape : DrawingShape
    {
        internal Point2F point0, point1;

        public LineShape(RenderTarget initialRenderTarget, Random random, D2DFactory d2DFactory, D2DBitmap bitmap)
            : base(initialRenderTarget, random, d2DFactory, bitmap)
        {
            point0 = RandomPoint();
            point1 = RandomPoint();
            PenBrush = RandomBrush();
            StrokeWidth = RandomStrokeWidth();
            if (CoinFlip)
                StrokeStyle = RandomStrokeStyle();
        }

        protected internal override void ChangeRenderTarget(RenderTarget newRenderTarget)
        {
            PenBrush = CopyBrushToRenderTarget(PenBrush, newRenderTarget);
        }

        protected internal override void Draw(RenderTarget renderTarget)
        {
            if (StrokeStyle != null)
                renderTarget.DrawLine(point0, point1, PenBrush, StrokeWidth, StrokeStyle);
            else
                renderTarget.DrawLine(point0, point1, PenBrush, StrokeWidth);
        }
    }
}

File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/VB/Direct2D/D2DShapes/LineShape.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.

Imports Microsoft.VisualBasic
Imports System
Imports Microsoft.WindowsAPICodePack.DirectX.Direct2D1

Namespace D2DShapes
	Friend Class LineShape
		Inherits DrawingShape
		Friend point0, point1 As Point2F

		Public Sub New(ByVal initialRenderTarget As RenderTarget, ByVal random As Random, ByVal d2DFactory As D2DFactory, ByVal bitmap As D2DBitmap)
			MyBase.New(initialRenderTarget, random, d2DFactory, bitmap)
			point0 = RandomPoint()
			point1 = RandomPoint()
			PenBrush = RandomBrush()
			StrokeWidth = RandomStrokeWidth()
			If CoinFlip Then
				StrokeStyle = RandomStrokeStyle()
			End If
		End Sub

		Protected Friend Overrides Sub ChangeRenderTarget(ByVal newRenderTarget As RenderTarget)
			PenBrush = CopyBrushToRenderTarget(PenBrush, newRenderTarget)
		End Sub

		Protected Friend Overrides Sub Draw(ByVal renderTarget As RenderTarget)
			If StrokeStyle IsNot Nothing Then
				renderTarget.DrawLine(point0, point1, PenBrush, StrokeWidth, StrokeStyle)
			Else
				renderTarget.DrawLine(point0, point1, PenBrush, StrokeWidth)
			End If
		End Sub
	End Class
End Namespace
